fs/operations/dedupe.go

```diff
@@ -245,6 +250,70 @@ func dedupeMergeDuplicateDirs(ctx context.Context, f fs.Fs, duplicateDirs [][]fs
 	return nil
 }
 
+// findLargestDirectory finds the largest directory based on collective size of contents and returns index of largestDirectory
+func findLargestDirectory(ctx context.Context, f fs.Fs, dirs []fs.Directory) (largestDirIndex int, err error) {
+	var dirSizes []int64
+	var wg sync.WaitGroup
+	var muErr sync.RWMutex
+
+	for i, dir := range dirs {
+		go func() {
+			wg.Add(1)
+			defer wg.Done()
+			muErr.RLock()
+			if err != nil {
+				muErr.RUnlock()
+				return
+			}
+			muErr.RUnlock()
+			dirSize, localErr := directorySize(ctx, f, dir, true)
+			if localErr != nil {
+				muErr.Lock()
+				err = errors.Wrap(localErr, "error in finding size of directory")
+				muErr.Unlock()
+				return
+			}
+			dirSizes[i] = dirSize
+		}()
+	}
+
+	wg.Wait()
+	var largestDirSize int64
+	for i, _ := range dirSizes {
+		if dirSizes[i] > largestDirSize {
+			largestDirSize = dirSizes[i]
+			largestDirIndex = i
+		}
+	}
+	return
+}
+
+// directorySize returns the size of directory. includes size of subDirectories if recursive == true
+func directorySize(ctx context.Context, f fs.Fs, dir fs.Directory, recursive bool) (size int64, err error) {
```

Comment:
I note that there is a function which does pretty much exactly this already.  It would need a dir parameter added.

https://github.com/rclone/rclone/blob/18c029e0f0c3a5dd0c925e285ce39071cb8728ff/fs/operations/operations.go#L1032

Reply:
I think it's impossible to satisfy your `directorySize` request with existing `fs.Fs` interfaces. See below.

Reply:
Solved it differently. See my comment below.